sw/tools/generators/gen_flight_plan.ml

```diff
@@ -324,7 +324,7 @@ let rec print_stage = fun index_of_waypoints x ->
     match String.lowercase (Xml.tag x) with
         "return" ->
           stage ();
-          lprintf "Return();\n";
+          lprintf "Return(%s);\n" (ExtXml.attrib x "reset_stage");
```

Comment:
You should not rely on the DTD to be sure that `reset_stage` attribute will exist. Better use `ExtXml.attrib_or_default` function instead.